specs/language/introduction.tex

```diff
@@ -77,46 +118,67 @@
+execute \acrshort{hlsl} programs on a supported \textit{device}. There is no
+strict requirement that the host and device be different physical hardware,
+although they commonly are. The separation of host and device in this
+specification is useful for defining the execution and memory model as well as
+specific semantics of language constructs.
+
+\Par{\gls{lane}}{Intro.Model.Terms.Lane}
+A \gls{lane} represents a single computed element in an \acrshort{spmd}
 program. In a traditional programming model it would be analogous to a thread of
 execution, however it differs in one key way. In multi-threaded programming
 threads advance independent of each other. In \acrshort{spmd} programs, a group
 of \gls{lane}s execute instructions in lock step because each instruction is a
 \acrshort{simd} instruction computing the results for multiple \gls{lane}s
 simultaneously.
 
-\Sub{\gls{wave}}{Intro.Model.Wave}
-
-\p A grouping of \gls{lane}s for execution is called a \gls{wave}. \gls{wave}
-sizes vary by hardware architecture. Some hardware implementations support
-multiple wave sizes. Generally wave sizes are powers of two, but there is no
-requirement that be the case. \acrshort{hlsl} is explicitly designed to run on
-hardware with arbitrary \gls{wave} sizes.
-
-\Sub{\gls{quad}}{Intro.Model.Quad}
-
-\p A \gls{quad} is a subdivision of four \gls{lane}s in a \gls{wave} which are
+\Par{\gls{wave}}{Intro.Model.Terms.Wave} A grouping of \gls{lane}s for execution
+is called a \gls{wave}. \gls{wave} sizes vary by hardware architecture. Some
+hardware implementations support multiple wave sizes. Generally wave sizes are
+powers of two, but there is no requirement that be the case. \acrshort{hlsl} is
+explicitly designed to run on hardware with arbitrary \gls{wave} sizes. Hardware
+architectures may implement \gls{wave}s as \acrfull{simt} where each thread
+executes instructions in lock-step. This is not a requirement of the model. Some
+constructs in \acrshort{hlsl} require synchronized execution. Such constructs
+will explicitly specify that requirement.
+
+\Par{\gls{quad}}{Intro.Model.Terms.Quad}
+A \gls{quad} is a subdivision of four \gls{lane}s in a \gls{wave} which are
 computing adjacent values. In pixel shaders a \gls{quad} may represent four
 adjacent pixels and \gls{quad} operations allow passing data between adjacent
 lanes. In compute shaders quads may be one or two dimensional depending on the
 workload dimensionality described in the \texttt{numthreads} attribute on the
 entry function. (FIXME: Add reference to attribute)
 
-\Sub{\gls{threadgroup}}{Intro.Model.Group}
+\Par{\gls{threadgroup}}{Intro.Model.Terms.Group}
+A grouping of \gls{lane}s executing the same shader to produce a combined
+result is called a \gls{threadgroup}. \gls{threadgroup}s are independent of
+\acrshort{simd} hardware specifications. The dimensions of a \gls{threadgroup}
+are defined in three dimensions. The maximum dimensions of a \gls{threadgroup}
```

Comment:
wording: I suggest "the maximum extent along each dimension of a ..." because "maximum dimensions" here is just 3.